06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.index = {};
+    this.object = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+  }
+  /**
+   * Gets Indexes
```

Comment:
Does the `createIndex` actually get the indices created or create them in the first instance. How about having a method that helps get indices that have been created? Because from this I sense you'd be accessing the instance variables directly from your controller.